sources/osgPlugins/ReaderWriterGLTF.js

```diff
@@ -0,0 +1,1060 @@
+    },
+
+    findFileFromURI: function ( urlOrFiles, uri ) {
+
+        if ( typeof ( urlOrFiles ) === 'string' ) return uri;
+
+        for ( var i = 0; i < urlOrFiles.length; ++i ) {
+
+            var fileName = this.getFileName( urlOrFiles[ i ] );
+
+            if ( fileName === uri )
+                return urlOrFiles[ i ];
+
+        }
+
+        return null;
+    },
+
+    getFileType: function ( urlOrFile ) {
+
+        var fileName = this.getFileName( urlOrFile );
+        if ( !fileName ) return null;
+
+        var ext = fileName.split( '.' ).pop();
+        if ( ext === 'bin' )
+            return 'arraybuffer';
+        else if ( ext === 'gltf' )
+            return 'text';
+
+        return 'blob';
+    },
+
+    loadGLTFJson: function ( urlOrFiles ) {
+
+        var glTFFileOrUrl = null;
+        if ( typeof ( urlOrFiles ) === 'string' ) {
+
+            var ext = urlOrFiles.split( '.' ).pop();
+            if ( ext !== 'gltf' ) {
+
+                console.warn( 'The given URL does not point toward a valid glTF file' );
+                return Promise.resolve( null );
+
+            }
+
+            var index = urlOrFiles.lastIndexOf( '/' );
+            this._localPath = ( index === -1 ) ? '' : urlOrFiles.substr( 0, index + 1 );
+
+            glTFFileOrUrl = urlOrFiles;
+
+        } else {
+
+            if ( !( urlOrFiles instanceof FileList ) ) {
+
+                console.warn( 'The provided argument is neither a files array nor a valid URL' );
+                return Promise.resolve( null );
+
+            }
+
+            for ( var i = 0; i < urlOrFiles.length; ++i ) {
+
+                var fileName = this.getFileName( urlOrFiles[ i ] );
+
+                if ( fileName.split( '.' ).pop() === 'gltf' ) {
+
+                    glTFFileOrUrl = urlOrFiles[ i ];
+                    break;
+
+                }
+
+            }
+
+        }
+
+        if ( !glTFFileOrUrl ) {
+
+            console.warn( 'You did not provided any glTF file' );
+            return Promise.resolve( null );
+
+        }
+
+        return this.loadFile( glTFFileOrUrl, 'gltf' );
+
+    },
+
+    requestFileFromReader: function ( file, fileType ) {
+
+        var defer = Promise.defer();
+
+        if ( !( file instanceof File ) )
+            return defer.reject();
+
+        var reader = new window.FileReader();
+        reader.onload = function ( data ) {
+
+            if ( fileType !== 'blob' )
+                defer.resolve( data.target.result );
+            else {
+                var img = new window.Image();
+                img.src = data.target.result;
+
+                defer.resolve( img );
+            }
+
+        };
+
+        if ( fileType === 'arraybuffer' )
+            reader.readAsArrayBuffer( file );
+        else if ( fileType === 'text' )
+            reader.readAsText( file );
+        else
+            reader.readAsDataURL( file );
+
+        return defer.promise;
+    },
+
+    requestImageURL: function ( url ) {
```

Comment:
It looks very similar to Input.readImageURL, so probably we could use the same code. Could need a bit of refactoring but I think it would be better to have only one method to fetch data type